test/ng/compileSpec.js

```diff
@@ -4827,6 +4827,31 @@ describe('$compile', function() {
             });
         });
 
+
+        it('should reassign bindings to the controller before $onInit, in case the constructor has overwritten them', function() {
+          angular.module('owComponentTest', [])
+            .component('owComponent', {
+              bindings: { input: '<' },
+              controller: function() {
+                component = this;
+                this.input = 'constructor-changed';
+                this.$onInit = function() {
+                  log.push('$onInit: input=' + this.input);
+                };
+              }
+            });
+
+          module('owComponentTest');
+          inject(function() {
+            $rootScope.name = 'outer';
+            compile('<ow-component input="name"></ow-component>');
+            expect(component.input).toEqual('outer');
```

Comment:
Because currently the properties are supposed to be initialized _before_ instantiating the constructor (so that they are available in the constructor function. The user can use the values, but also overwrite them.

With this change, we are overwriting them again after the constructor function has been called.
This is a BC imo.